gnocchi/carbonara.py

```diff
@@ -597,54 +608,49 @@ def split(self):
         # this is slow because pandas can do that on any kind DataFrame
         # but we have ordered timestamps, so don't need
         # to iter the whole series.
-        freq = self.sampling * SplitKey.POINTS_PER_SPLIT
+        freq = self.aggregation.granularity * SplitKey.POINTS_PER_SPLIT
         keys, counts = numpy.unique(
             round_timestamp(self.timestamps, freq),
             return_counts=True)
         start = 0
         for key, count in six.moves.zip(keys, counts):
             end = start + count
-            yield (SplitKey(key, self.sampling),
-                   AggregatedTimeSerie(self.sampling, self.aggregation_method,
-                                       self[start:end]))
+            yield (SplitKey(key, self.aggregation.granularity),
+                   AggregatedTimeSerie(self.aggregation, self[start:end]))
             start = end
 
     @classmethod
-    def from_timeseries(cls, timeseries, sampling, aggregation_method):
+    def from_timeseries(cls, timeseries, aggregation):
```

Comment:
just for discussion, there are places where we clearly require sampling to be defined but as we start passing in Aggregation object, we lose the ability to enforce this. any thoughts how we can handle this? are we relying completely on tests to validate this?

Reply:
I don't think this patch broke any existing enforcement, so AFAIU it's already the case that we solely rely on the tests for this. If you have a more precise example I may have a more precise answer. :)

Reply:
works for me. let's see if it comes back to bite us